addons/hr_expense/static/src/components/attachment_view.js

```diff
@@ -0,0 +1,14 @@
+/** @odoo-module **/
+
+import { patch } from "@web/core/utils/patch";
+import { AttachmentView } from "@mail/core/common/attachment_view";
+
+patch(AttachmentView.prototype, {
+    displayName() {
+        if (this.state.thread.model === 'hr.expense.sheet') {
+            return (this.state.thread.mainAttachment.res_name || this.state.thread.name) + ' - ' + this.state.thread.mainAttachment.filename
+        } else {
+            return this.state.thread.mainAttachment.filename
+        }
```

Comment:
You almost always need to call super when inheriting in Odoo :wink: 
```suggestion
        }
        return this._super()
```

Reply:
Seems not to be working but "super.displayName" works